File: modules/angular2/test/symbol_inspector/simple_library.ts
// A library for the symbol inspector test

export class A {
  constructor(b: ConsParamType) {}

  field: FieldType;
  get getter(): GetterType { return null; }

  method(p: ParamType): MethodReturnType { return null; }

  methodWithFunc(closure: ClosureReturn) {}

  static staticField: StaticFieldType = null;
  static staticMethod() {}
}

export class ConsParamType {}
export class FieldType {}
export class GetterType {}
export class MethodReturnType {}
export class ParamType {}
export class StaticFieldType {}

export class ClosureReturn {}
export class ClosureParam {}

export class TypedefReturnType {}
export class TypedefParam {}


export class Generic<K> {  // Generic should be exported, but not K.
  get getter(): K { return null; }
}

export interface SomeInterface { someMethod(); }

File: test/symbol_inspector/simple_library.dart
// A library for the symbol inspector test
library angular2.test.symbol_inspector.simple_library;

class A {
  A(ConsParamType b) {}
  FieldType field;
  GetterType get getter {
    return null;
  }

  MethodReturnType method(ParamType p) {
    return null;
  }

  methodWithFunc(ClosureReturn closure) {}
  static StaticFieldType staticField = null;
  static staticMethod() {}
}

class ConsParamType {}

class FieldType {}

class GetterType {}

class MethodReturnType {}

class ParamType {}

class StaticFieldType {}

class ClosureReturn {}

class ClosureParam {}

class TypedefReturnType {}

class TypedefParam {}

class Generic<K> {
  K get getter {
    return null;
  }
}

abstract class SomeInterface {
  someMethod();
}
